src/interface/guide/foundation/FoundationDowntimeSectionV2.tsx

```diff
@@ -0,0 +1,513 @@
+              <>
+                <dt>
+                  <PerformanceStrong performance={melee.meleeUptimePerformance}>
+                    {formatPercentage(melee.meleeUptimePercentage, 1)}%
+                  </PerformanceStrong>{' '}
+                </dt>
+                <dd>
+                  <TooltipElement content="The percentage of time that your basic melee swings were active, excluding time spent casting.">
+                    Melee Uptime
+                  </TooltipElement>
+                </dd>
+              </>
+            )}
+          </UptimeStatistics>
+        </div>
+        <div>
+          <Para>
+            The foundation of good play in <em>WoW</em> is having good <HL>uptime.</HL>{' '}
+            <ByRole>
+              <Role.Melee>
+                There should be no gaps between the end of one <GCD /> and the start of the next.
+              </Role.Melee>
+              <Role.Caster>
+                There should be no gaps between the end of one spell cast and the start of the next.
+              </Role.Caster>
+            </ByRole>{' '}
+            This diagram shows gaps in your uptime in{' '}
+            <Highlight color={BadColor} textColor="white">
+              red
+            </Highlight>
+            .
+          </Para>
+          <Para>
+            With practice, you will be able to maintain uptime <em>and</em> pick the right abilities
+            for each moment, but remember that{' '}
+            <strong>doing something is better than doing nothing</strong>.
+          </Para>
+        </div>
+      </div>
+      <SubSection>
+        <ComplexUptimeDisplay
+          uptimeHistory={uptimeHistory}
+          meleeGaps={melee?.meleeUptimeGaps}
+          globalMeleeGaps={globalMeleeUptime}
+          debuffSegments={debuffs?.debuffSegments}
+        />
+      </SubSection>
+      <SubSection>
+        <ul className="list issues">
+          <SmallGapsSuggestion />
+        </ul>
+      </SubSection>
+    </>
+  );
+}
+
+interface Props {
+  uptimeHistory: Segment[];
+  meleeGaps?: Array<Segment>;
+  globalMeleeGaps?: Array<Segment>;
+  debuffSegments?: Array<DisplaySegment>;
+}
+
+interface Segment {
+  start: number;
+  end: number;
+}
+
+const UptimeStatistics = styled.dl`
+  display: grid;
+  grid-template-columns: max-content max-content;
+  grid-gap: 0 0.75em;
+  font-size: 2rem;
+  align-items: baseline;
+
+  & dd {
+    font-size: 1.7rem;
+    opacity: 80%;
+
+    & dfn {
+      border-bottom: unset;
+      text-decoration: underline;
+      text-decoration-style: dotted;
+    }
+  }
+`;
+
+function ComplexUptimeDisplay({
+  uptimeHistory,
+  meleeGaps,
+  globalMeleeGaps,
+  debuffSegments: rawDebuffSegments,
+}: Props): JSX.Element | null {
+  const info = useInfo();
+  const { fight } = useFight();
+
+  const boss = findByBossId(fight.boss);
+  const debuffSegments = useMemo(
+    () => rawDebuffSegments?.filter((segment) => segment.end - segment.start >= 4000) ?? [],
```

Comment:
What's the significance of 4000 here? Maybe put it into a constant with a detailed name

Reply:
just an arbitrary threshold to cut out very short debuffs. it is a bit of an artifact of when i was trying to pick debuffs automatically instead of just using a list and can probably be removed entirely.